doc/dev/adr/0027-generic-usage-of-cloud-event-headers.md

```diff
@@ -38,21 +37,78 @@ public class CloudEvent
     public string? DataContentType;
     public string? Subject;
     public string? DataSchema;
+}
+
+public class MqttApplicationMessage 
+{
+  //...
+
+  public void SetCloudEvent(CloudEvent cloudEvent);
+
+  public CloudEvent? GetCloudEvent()
 
-    // The two new APIs for converting to and from Mqtt user properties/Mqtt content type
-    public CloudEventMqttContext CreateMqttMessageContext();
-    public static CloudEvent CreateFromMqttUserProperties(CloudEventMqttContext mqttMessageContext);
+  //...
 }
 
-public class CloudEventMqttContext
+```
+
+### Example usage
+
+#### Sending side
+
+```csharp
+public static void main()
 {
-    public List<MqttUserProperty> MqttUserProperties;
-    public string MqttMessageContentType;
+    MqttClient client = new ...;
+    client.ConnectAsync(...);
+
+    CloudEvent cloudEvent = new CloudEvent()
+    {
+      Source = ...
+      Type = ...
+      DataSchema = ...
+    };
+
+    MqttApplicationMessage mqttMessage = new(...);
+
+    // The appropriate headers are set and the content type is set
+    mqttMessage.SetCloudEvent(cloudEvent);
```

Comment:
How does this resolve if there's already a content type set on the mqttMessage?

Reply:
It will need to override the already set content type. I don't think we have much choice there since the cloud event spec expects it

Reply:
I'll call this out in the ADR